distributed/shuffle/tests/test_shuffle.py

```diff
@@ -707,16 +798,16 @@ def test_processing_chain():
     filesystem = defaultdict(io.BytesIO)
 
     for partitions in splits_by_worker.values():
-        for partition, batches in partitions.items():
-            for batch in batches:
-                dump_batch(batch, filesystem[partition], schema)
+        for partition, tables in partitions.items():
+            dump_shards(tables, filesystem[partition])
 
     out = {}
     for k, bio in filesystem.items():
         bio.seek(0)
-        out[k] = load_arrow(bio)
+        out[k] = load_partition(bio)
 
     assert sum(map(len, out.values())) == len(df)
+    assert all(v.to_pandas().dtypes.equals(df.dtypes) for v in out.values())
```

Comment:
(Not meant as a blocking comment) We're checking lengths and dtypes here. Maybe we could just use `pd.testing.assert_frame_equal` instead?